sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientTest.java

```diff
@@ -315,26 +325,85 @@ public void signDataVerifyRsa(HttpClient httpClient, CryptographyServiceVersion
+            if (isCurveSupportedByRuntime(signVerifyEcData.getCurve())) {
+                try {
+                    String algorithmName = "EC";
+                    Provider[] providers = Security.getProviders();
+                    Provider provider = null;
 
-            byte[] plainText = new byte[100];
+                    for (Provider currentProvider : providers) {
+                        if (currentProvider.containsValue(algorithmName)) {
+                            provider = currentProvider;
 
-            new Random(0x1234567L).nextBytes(plainText);
+                            break;
+                        }
+                    }
 
-            byte[] signature =
-                cryptographyClient.signData(curveToSignature.get(curve), plainText).getSignature();
-            Boolean verifyStatus =
-                cryptographyClient.verifyData(curveToSignature.get(curve), plainText, signature).isValid();
+                    if (provider == null) {
+                        for (Provider currentProvider : providers) {
+                            System.out.println(currentProvider.getName());
+                        }
 
-            assertTrue(verifyStatus);
+                        fail(String.format("No suitable security provider for algorithm %s was found.", algorithmName));
+                    }
+
+                    final KeyPairGenerator generator = KeyPairGenerator.getInstance(algorithmName, provider);
+                    ECGenParameterSpec spec =
+                        new ECGenParameterSpec(signVerifyEcData.getCurveToSpec().get(signVerifyEcData.getCurve()));
+
+                    generator.initialize(spec);
+
+                    KeyPair keyPair = generator.generateKeyPair();
+
+                    JsonWebKey jsonWebKey =
+                        JsonWebKey.fromEc(keyPair, provider, Arrays.asList(KeyOperation.SIGN, KeyOperation.VERIFY));
+                    KeyCurveName curve = signVerifyEcData.getCurve();
+                    Map<KeyCurveName, SignatureAlgorithm> curveToSignature = signVerifyEcData.getCurveToSignature();
+                    CryptographyClient cryptographyClient = initializeCryptographyClient(jsonWebKey);
+
+                    byte[] plainText = new byte[100];
+
+                    new Random(0x1234567L).nextBytes(plainText);
+
+                    byte[] signature =
+                        cryptographyClient.signData(curveToSignature.get(curve), plainText).getSignature();
+                    Boolean verifyStatus =
+                        cryptographyClient.verifyData(curveToSignature.get(curve), plainText, signature).isValid();
+
+                    assertTrue(verifyStatus);
+                } catch (InvalidAlgorithmParameterException | NoSuchAlgorithmException e) {
+                    fail(e);
+                }
+            }
         });
     }
 
+    private boolean isCurveSupportedByRuntime(KeyCurveName keyCurveName) {
```

Comment:
From discussions we had elsewhere, let's not check the Java runtime version and just try/catch.